### PR TITLE
Fixes result list in mobile and desktop

### PR DESCRIPTION
**Pull request by gkozlowaski**

### CODE REVIEW

File: src/components/SearchBar/SearchBar.js [hunk 1]
```diff
@@ -117,14 +120,11 @@ class SearchBar extends React.Component {
...
-      <div
-        className="SearchBar"
-        data-is-visible={visible}
-        onClick={this.handleClick}
-      >
+      <div className="SearchBar" data-is-visible={visible} onTouchTap={this.handleTouchTap}>
         <div
           className="SearchBar__input-container"
-          onClick={(e) => e.stopPropagation()}
+          onTouchTap={ (e) => { e.stopPropagation() } }
```
Comment (amoreira) [inline src/components/SearchBar/SearchBar.js]: @thor `onTouchTap` is obsolete, correct?

Comment (thor) [inline src/components/SearchBar/SearchBar.js]: @amoreira if I remember correctly, yes -- but I'm not the Thor you seek.

Comment (amoreira) [inline src/components/SearchBar/SearchBar.js]: @thor thank you anyway!

File: src/components/SearchBar/SearchBar.js [hunk 2]
```diff
@@ -144,34 +167,40 @@ class SearchBar extends React.Component {
...
-              }
-            </ul>
-          </div>
         </div>
+        {
+          canDisplay ?
+            <div
+              className="SearchResultList__wrapper"
+              data-has-results={canDisplay}
+              style={ style }
```
Comment (tamorim) [inline src/components/SearchBar/SearchBar.js]: @gkozlowaski Try to be consistent on the coding style. Use brackets with spaces or without, but don't mix both for the same purpose, please.

File: src/components/SearchBar/SearchBar.js [hunk 3]
```diff
@@ -22,15 +22,36 @@ class SearchBar extends React.Component {
...
+    });
+  }
+
+  componentDidMount() {
+    window.addEventListener('resize', this.handleResize);
   }
 
   componentDidUpdate() {
+    if (this.props.visible && this.state.windowWidth <= 991) {
+      this.props.pageScrollOff();
```
Comment (tamorim) [inline src/components/SearchBar/SearchBar.js]: @gkozlowaski I've commented about these props on the `theme`PR.

**Comment by tamorim** [comment]
@gkozlowaski Can I review it?

**Comment by gkozlowaski** [comment]
@tamorim yes, please!

File: src/components/SearchBar/SearchBar.js [hunk 4]
```diff
@@ -22,15 +22,39 @@ class SearchBar extends React.Component {
...
+      bodyOverflow: document.body.style.overflow
+    });
+  }
+
+  componentDidMount() {
+    window.addEventListener('resize', this.handleResize);
   }
 
   componentDidUpdate() {
+    if (this.props.visible && this.state.windowWidth <= 991) {
```
Comment (tamorim) [inline src/components/SearchBar/SearchBar.js]: @gkozlowaski What `991` represents? You use it more than once on the component, but I don't know why you picked this number, and what does it mean.
Now it's just a magic number!
You should use this value on a constant with a more meaningful name, like:
`const RESULT_LIST_HEIGHT = 991` (it's just a guess, again, I don't know what it means).

The first tip works if it's really necessary that kind of check... This work on all viewports? It seems a very specific value that may cause trouble.

File: src/components/SearchBar/SearchBar.js [hunk 5]
```diff
@@ -70,23 +94,21 @@ class SearchBar extends React.Component {
...
+  handleClick = () => {
     this.searchInput.value = '';
-    this.props.handleSearchClick();
-  }
-
-  handleClick = (e) => {
-    e.preventDefault();
-    this.searchInput.value = '';
-    this.props.handleSearchClick();
+    this.props.handleSearchTap();
```
Comment (tamorim) [inline src/components/SearchBar/SearchBar.js]: @gkozlowaski This one comes from a bad rebase, even the diff shows it.
The correct prop name is `handleSearchClick`. That's why the `x` button doesn't work on your branch.

**Comment by tamorim** [comment]
@gkozlowaski Done reviewing, check the comments

**Pull request merged**